packages/tupaia-web/src/utils/useModal.ts

```diff
@@ -13,8 +14,11 @@ export const useModal = () => {
   function navigateToModal(hashKey: `${MODAL_ROUTES}`) {
     navigate({ ...location, hash: hashKey });
   }
-  function closeModal() {
```

Comment:
I think there are scenarios where the modal needs to be closed but the url query string params need to be retained.  Would it be possible to keep the close modal as it was and do the removing of params in the onSuccess handler of the specific form that you are trying to clear the query string param for?

Reply:
So this is for when you close the modal by clicking out of the modal or clicking outside of it. And this one will only remove any params you pass in to be removed, e.g. the password reset token or the `project=` param

Reply:
Now that I've read the rest of the code I can see how this is working now. It only removes the passed in params so should work for all cases. You can disregard the comment above 🙏